src/main/java/io/github/mfoo/libyear/LibYearMojo.java

```diff
@@ -496,6 +524,61 @@ public void execute() throws MojoExecutionException {
         }
     }
 
+    private void generateReport(Set<Dependency> dependencies, final String categorie) {
+        if (StringUtils.isBlank(reportFile)) {
+            return;
+        }
+
+        StringBuilder logsToReport = new StringBuilder();
+
+        dependencies.stream().forEach(dependency -> {
+            try {
+                Artifact artifact = getHelper().createDependencyArtifact(dependency);
+
+                Optional<LocalDate> currentReleaseDate = getReleaseDate(
+                        artifact.getGroupId(),
+                        artifact.getArtifactId(),
+                        artifact.getSelectedVersion().toString());
+
+                String depName = dependency.getGroupId() + ":" + dependency.getArtifactId();
+                String libYearsStr = "unknown";
+                if (!currentReleaseDate.isEmpty()) {
+                    long libWeeksOutdated = ChronoUnit.WEEKS.between(currentReleaseDate.get(), LocalDate.now());
+                    float libYearsOutdated = libWeeksOutdated / 52f;
+
+                    if (libYearsOutdated > 0
+                            && (minLibYearsForReport <= 0 || libYearsOutdated > minLibYearsForReport)) {
+                        libYearsStr = String.format(Locale.US, "%.2f", libYearsOutdated);
+                    }
+                }
+                logsToReport
+                        .append(depName)
+                        .append(",")
+                        .append(dependency.getVersion())
+                        .append(",")
+                        .append(dependency.getType())
+                        .append(",")
+                        .append(categorie)
+                        .append(",")
+                        .append(libYearsStr)
+                        .append(System.lineSeparator());
+            } catch (MojoExecutionException | OverConstrainedVersionException e) {
+                getLog().error("Exception by writing report", e);
+            }
+        });
+
+        File targetDirectory = new File(project.getModel().getBuild().getDirectory());
+        if (targetDirectory.exists() && targetDirectory.isDirectory()) {
+            Path path = Paths.get(targetDirectory.getAbsolutePath(), reportFile);
+            try {
+                Files.write(
+                        path, logsToReport.toString().getBytes(), StandardOpenOption.CREATE, StandardOpenOption.APPEND);
```

Comment:
Probably minor thing here. If we run the plugin multiple times without a `clean`, it will get longer each time:

```sh
$ mvn io.github.mfoo:libyear-maven-plugin:1.1.1-SNAPSHOT:analyze -D reportFile=foo.csv > /dev/null
$ wc -l target/foo.csv
26 target/foo.csv
$ mvn io.github.mfoo:libyear-maven-plugin:1.1.1-SNAPSHOT:analyze -D reportFile=foo.csv > /dev/null
$ wc -l target/foo.csv
52 target/foo.csv
```

If we changed this to make a new file, though, then the function wouldn't report properly as it's called multiple times. Since this PR has been waiting for me so long, I'm inclined to get it merged now and consider that later.